nav2_costmap_2d/include/nav2_costmap_2d/costmap_2d_ros.hpp

```diff
@@ -140,6 +140,14 @@ class Costmap2DROS : public nav2_util::LifecycleNode
    */
   bool getRobotPose(geometry_msgs::msg::PoseStamped & global_pose);
 
+  /**
+   * @brief Transform the pose in the global frame of the costmap
+   * @param pose will be set to the pose of the robot in the global frame of the costmap
+   * @return True if the pose was transformed successfully, false otherwise
+   */
+
+  bool getPoseInGlobalFrame(geometry_msgs::msg::PoseStamped & pose);
```

Comment:
This feels weird to me to be in the costmap, why not put this in the planner server? This name also confuses me a bit, you're really just transforming a given pose into the global frame, but the `get` is just odd as if the input is an empty pose, but really its a meaningful pose. Something about this just doesn't quite register right in my mind, do you have that feeling too?

Reply:
> This feels weird to me to be in the costmap, why not put this in the planner server?

In order to avoid to create a tf_buffer and a tf_listener in the planner server just for this. And since the frame in which we want the pose to be relative to is the global_frame of the costmap.
Also because I followed the logic of `getRobotPose`



> Something about this just doesn't quite register right in my mind, do you have that feeling too?

I do too. Let's find a better name, few propositions but I cannot find a satisfactory one:  `transformPoseInGlobalFrame`, `referencePoseRelativeToGlobalFrame`, `putPoseInCostmapFrame` ?

Reply:
https://github.com/ros-planning/navigation2/blob/main/nav2_planner/include/nav2_planner/planner_server.hpp#L135 the planner server already has the buffer/listener from the costmap itself. I'd probably be OK with this if the name / API of this function felt "more right" to us, and I know that's a really wishy-washy thing to say :laughing: 

The things that are really throwing me for a loop
- the `get` word when you're actually inputting a valid non-identity pose, so its not just "get" but "transform"ing
- The fact that the input and output are the same variable and remapped over each other

I think `transformPoseToGlobalFrame` would do.

Reply:
Ok, I changed the names for `transformPoseToGlobalFrame` and `transformPoseInTargetFrame` + ` const input_pose` and `transformed_pose` parameter pattern.
If you are ok with that, let's leave the discussion about the best place to call the nav2_utils fonctions (ie. planner_server or costmap) for another day.